src/t8_schemes/t8_standalone/t8_standalone_implementation.hxx

```diff
@@ -0,0 +1,1568 @@
+   *                      lies within the root tree.
+   * \param [in] root_face The index of the face of the root tree in which \a face
+   *                      lies.
+   * \return              The face number of the face of \a elem that coincides
+   *                      with \a face.
+   */
+  static inline int
+  element_extrude_face (const t8_element_t *face, t8_element_t *elem, int root_face, const t8_scheme *face_scheme)
+  {
+    SC_ABORT ("This function is not implemented in this scheme yet.\n");
+    return 0;
+  }
+
+  /** Construct the boundary element at a specific face.
+   * \param [in] elem     The input element.
+   * \param [in] face     The index of the face of which to construct the
+   *                      boundary element.
+   * \param [in,out] boundary An allocated element of dimension of \a element
+   *                      minus 1. The entries will be filled with the entries
+   *                      of the face of \a element.
+   * \param [in] boundary_scheme The scheme for the eclass of the boundary face.
+   * If \a elem is of class T8_ECLASS_VERTEX, then \a boundary must be NULL
+   * and will not be modified.
+   */
+  static inline void
+  element_get_boundary_face (const t8_element_t *elem, int face, t8_element_t *boundary,
+                             const t8_scheme *boundary_scheme)
+  {
+    SC_ABORT ("This function is not implemented in this scheme yet.\n");
+  }
+
+  // ################################################____LINEAR ID____################################################
+
+  /** Initialize the entries of an allocated element according to a
+   *  given linear id in a uniform refinement.
+   * \param [in,out] elem The element whose entries will be set.
+   * \param [in] level    The level of the uniform refinement to consider.
+   * \param [in] id       The linear id.
+   *                      id must fulfil 0 <= id < 'number of leaves in the uniform refinement'
+   */
+  static inline void
+  element_set_linear_id (t8_element_t *elem, int level, t8_linearidx_t id)
+  {
+
+    t8_standalone_element<TEclass> *el = (t8_standalone_element<TEclass> *) elem;
+
+    t8_standalone_scheme<TEclass>::get_root ((t8_element_t *) el);
+
+    /* There is only one element at level 0, so it must be root */
+    if (level == 0) {
+      T8_ASSERT (id == 0);
+      return;
+    }
+
+    T8_ASSERT (0 <= id);
+    T8_ASSERT (1 <= level && level <= T8_ELEMENT_MAXLEVEL[TEclass]);
+
+    while (el->level < level) {
+      /* Shortcut if we need the first descendant of the subtree*/
+      if (id == 0) {
+        t8_standalone_scheme<TEclass>::element_get_first_descendant ((const t8_element_t *) el, (t8_element_t *) el,
+                                                                     level);
+        return;
+      }
+
+      t8_linearidx_t sum_descendants_of_children_before = 0;
+      t8_linearidx_t num_descendants_of_child = 0;
```

Comment:
```suggestion
```
The scope of num_descendants_of_child is actually inside the while loop so it can be removed here.